### ISSUE TITLE
Feature request: check that Map<K, V> is an EnumMap if K is an enum value

### ISSUE THREAD

**Issue by lgoldstein** [pmd/pmd#577]
Would be nice to have a rule that enforces this practice - the rationale being that it makes no sense to use `HashMap` or `TreeMap` when `EnumMap` is available whenever the keys are _Enum_(s)

**Comment by jsotuyod** [comment]
@lgoldstein sounds really interesting! With the current status of type resolution, after the great efforts from @WinterGrascph, this should be doable.

**Comment by oowekyala** [comment]
Yeah that's a good idea! Would be nice if it covered EnumSets too
